### ISSUE TITLE
[Bug]: Sonner component needs vanila JS support.

### ISSUE THREAD

**Issue by assasin0076** [radix-vue/shadcn-vue#313]
### Reproduction

Add sonner component to vue3 js project

### Describe the bug

![image](https://github.com/radix-vue/shadcn-vue/assets/38113445/9678116a-c5a3-47ee-a87d-46aa265805d4)
I got this error when i trying to add sonner component to the vanila js vue3 project.

### System Info

```bash
System:
    OS: Windows 10 10.0.19045
    CPU: (16) x64 AMD Ryzen 7 3800X 8-Core Processor
    Memory: 15.25 GB / 31.95 GB
  Binaries:
    Node: 21.5.0 - C:\Program Files\nodejs\node.EXE
    Yarn: 1.22.19 - ~\AppData\Roaming\npm\yarn.CMD
    npm: 9.6.6 - ~\AppData\Roaming\npm\npm.CMD
    pnpm: 7.21.0 - ~\AppData\Local\pnpm\pnpm.EXE
  Browsers:
    Chrome: master_preferences
    Edge: Chromium (120.0.2210.91)
    Internet Explorer: 11.0.19041.3636
```

### Contributes

- [ ] I am willing to submit a PR to fix this issue
- [ ] I am willing to submit a PR with failing tests

**Comment by hrynevychroman** [comment]
@assasin0076 am i right that every component is not usable in your project? If this is true, I think the best and ease approach to install `typescript` as a `devDependencies`. It is because of direct copy components to your project, we don't compile them to vanilla JS.

If it will help, I think the best will be to mention it inside Docs install sections.

**Comment by gregmsanderson** [comment]
@romanhrynevych I'm having the same issue with `sonner`. It's a JavaScript project (I set typescript to false in `components.json`, as documented).

As far as i can see the CLI _does_ convert the files to JS. I tried installing all the components. All but three work. For example after installing the Button, I get a `Button.vue` and sibling `index.js`, both of which are valid.

There are currently three that fail to install using the CLI, all for TypeScript-related issues (since the TypeScript code is not being removed):

- calendar
- sonner
- toast

All the rest of the components install successfully into a JavaScript project.
